### ISSUE TITLE
Support for KV store snapshot

### ISSUE THREAD

**Issue by jumaffre** [microsoft/CCF#1299]
It should be possible to take a snapshot of the KV store at a given version (or at the last globally committed version) and successfully apply it to another KV store.

Note that for public recovery, the state of the public tables in the snapshot should only be integrity protected.

**Comment by jumaffre** [comment]
As discussed with @ashamis, let's go for a snapshot serialisation scheme that is similar to the KV write sets for now as it means more code paths that we can re-use. We can then build a benchmark and iterate on the design later but this should unblock other snapshot-related tickets quicker.

**Comment by jumaffre** [comment]
I believe this is done as we can serialise/deserialise snaphots between store. As per my earlier comment, we can still optimise performance on this but I think raising a separate issue for it is preferable. @ashamis, feel free to close this ticket if you think this is done and I'll create a new ticket for performance improvement.
